docs/education/dlt-advanced-course/lesson_1_custom_sources_restapi_source_and_restclient.py

```diff
@@ -0,0 +1,1223 @@
+# /// script
+# dependencies = [
+#     "dlt[duckdb]",
+#     "numpy",
+#     "pandas",
+#     "sqlalchemy",
+# ]
+# ///
+
+import marimo
+
+__generated_with = "0.17.4"
+app = marimo.App()
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""# **Building custom sources with [dlt REST API source](https://dlthub.com/docs/devel/dlt-ecosystem/verified-sources/rest_api/basic) and [RESTClient](https://dlthub.com/docs/devel/general-usage/http/rest-client)** [![Open in molab](https://marimo.io/molab-shield.svg)](https://molab.marimo.io/github/dlt-hub/dlt/blob/master/docs/education/dlt-advanced-course/lesson_1_custom_sources_restapi_source_and_restclient.py) [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/dlt-hub/dlt/blob/master/docs/education/dlt-advanced-course/lesson_1_custom_sources_restapi_source_and_restclient.ipynb) [![GitHub badge](https://img.shields.io/badge/github-view_source-2b3137?logo=github)](https://github.com/dlt-hub/dlt/blob/master/docs/education/dlt-advanced-course/lesson_1_custom_sources_restapi_source_and_restclient.ipynb)"""
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""# New section""")
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""
+    # **Recap**
+
+    In the **[dlt Fundamentals](https://github.com/dlt-hub/dlthub-education/tree/main/courses/dlt_fundamentals_dec_2024)** course, we learned two primary ways to build sources for REST APIs:
+
+    1. **Using low-level dlt decorators** (`@dlt.source` and `@dlt.resource`) with [`RESTClient`](https://dlthub.com/docs/devel/general-usage/http/rest-client).
+    2. **Using the built-in [`rest_api` source](https://dlthub.com/docs/devel/dlt-ecosystem/verified-sources/rest_api/basic)** with declarative configuration.
+
+    ---
+    """)
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""
+    ### **1. Building sources with low-level dlt decorators**
+
+    We constructed a custom source for the **GitHub API** using the `RESTClient` class, decorators like `@dlt.resource` and `@dlt.source`, and manual pagination handling.
+
+
+    #### **Example**
+    """)
+    return
+
+
+@app.cell
+def _():
+    from typing import Iterator, Any, Iterable
+    import os
+    import dlt
+    from dlt.common.typing import TDataItems, TDataItem
+    from dlt.sources import DltResource
+    from dlt.sources.helpers import requests
+    from dlt.sources.helpers.rest_client import RESTClient
+    from dlt.sources.helpers.rest_client.auth import BearerTokenAuth
+    from dlt.sources.helpers.rest_client.paginators import HeaderLinkPaginator
+    os.environ['ACCESS_TOKEN'] = os.getenv('ACCESS_TOKEN')
```

Comment:
This doesn't seem to do anything (except fail if the environment variable is not set).
We should then have a prompt in markdown for the user to potentially set their `ACCESS_TOKEN` again